File: GroupAnagrams.js
function groupAnagrams(string)
{
    list1=[]
    for(i=0;i<string.length;i++)
    {
        list1.push(string[i].split("").sort().join(""))
    }
    obj={}
    for(i=0;i<string.length;i++)
    {
        if(list1[i] in obj)
        {
            obj[list1[i]].push(string[i])
            
        } 
        else
        {
            obj[list1[i]]=[string[i]]

        }

    }
    result=[]
    for(i in obj)
    {
        result.push(obj[i])
    }
    console.log(result)
}
string=["silent","listen","peacock"]
groupAnagrams(string)

File: GroupAnagrams.py
def groupAnagrmas(string):
    list1=[]
    for i in string:
        list1.append(''.join(sorted(i)))
    dict1={}
    for i,e in enumerate(list1):
        if e not in dict1:
            dict1[e]=[string[i]]
        else:
            dict1[e].append(string[i])
    li=[]
    for i,el in enumerate(dict1):
        li.append(dict1[el])
    print(li)
        
string=["abc","atc","peacock","tca","stf","listen","silent"]
groupAnagrmas(string)

